pkg/k8s/client.go

```diff
@@ -36,5 +36,8 @@ func EnsureObject(ctx context.Context, kubeClient client.Client, obj client.Obje
 	if err != nil {
 		return err
 	}
+
+	// hacky way to restore the GVK for the object after create corrupts it. didnt dig. not sure why?
```

Comment:
So calling create mess with gvk somehow? Very strange.

Reply:
yeah. this literally wasted 2 3 hours of my time the other day. very strange. will come back to this once I unblock the gitea work.

Reply:
I wonder if we're applying some kind of object which hasnt had its gvk registered in our scheme?

Reply:
no, the object has its GVK up to the point we make the call to `create` it. it comes back from that call and it doesnt have the GVK in there anymore.